components/ai_chat/core/browser/utils.cc

```diff
@@ -178,6 +183,65 @@ const std::string& GetActionTypeQuestion(mojom::ActionType action_type) {
   return iter->second;
 }
 
+uint32_t GetMaxPageContentLength(std::optional<uint32_t> context_size) {
+  base::CheckedNumeric<uint32_t> result =
+      context_size.value_or(kCustomModelDefaultContextSize) *
+      kDefaultCharsPerToken;
```

Comment:
this should be:

```c++
base::CheckedNumeric<uint32_t> result = context_size.value_or(kCustomModelDefaultContextSize);
result *= kDefaultCharsPerToken;
```

Otherwise the multiplication op takes place before the assignment, and exterior to the `CheckedNumeric` instance.